### PR TITLE
Updates for 4.0

### PR DESCRIPTION
**Pull request by artulloss**
The API has changed and this plugin isn't working on the latest pmmp master branch. This fixes the plugin so it will run, however I added comments where the "fix" isn't faithful to the original code, because I can't seem to get an instance of the ChunkManager in the constructor.

### CODE REVIEW

File: src/muqsit/vanillagenerator/generator/VanillaGenerator.php [hunk 1]
```diff
@@ -18,6 +18,7 @@ abstract class VanillaGenerator extends Generator{
 
 	protected const WORLD_DEPTH = 128;
 
+	// TODO Figure out a non hacky way to use this. We don't have access to the ChunkManager necessary to call it in the constructor
```
Comment (Muqsit) [inline src/muqsit/vanillagenerator/generator/VanillaGenerator.php:21]: you may remove this, there is no non-hacky method to access a non-public property

File: src/muqsit/vanillagenerator/generator/biomegrid/RiverMapLayer.php [hunk 2]
```diff
@@ -5,12 +5,12 @@
...
 
 use muqsit\vanillagenerator\generator\overworld\biome\BiomeIds;
-use pocketmine\world\biome\Biome;
+use pocketmine\data\bedrock\BiomeIds as PMBiomeIds;
 
 class RiverMapLayer extends MapLayer{
 
 	/** @var int[] */
-	private static $OCEANS = [Biome::OCEAN => 0, BiomeIds::DEEP_OCEAN => 0];
+	private static $OCEANS = [PMBiomeIds::OCEAN => 0, BiomeIds::DEEP_OCEAN => 0];
```
Comment (Muqsit) [inline src/muqsit/vanillagenerator/generator/biomegrid/RiverMapLayer.php:13]: A better solution would be mapping the biome ID constants in `muqsit\vanillagenerator\generator\overworld\biome\BiomeIds` to the constants in `pocketmine\data\bedrock\BiomeIds`. VanillaGenerator's BiomeIds has more biome ids than pocketmine\data's. Its too messy to use both the classes together in every file.

File: src/muqsit/vanillagenerator/generator/nether/NetherGenerator.php [hunk 3]
```diff
@@ -41,9 +41,9 @@ private static function densityHash(int $i, int $j, int $k) : int{
 	/** @var int */
 	protected $bedrock_roughness = 5;
 
-	public function __construct(ChunkManager $world, int $seed, array $options = []){
-		parent::__construct($world, $seed, Environment::NETHER, null, $options);
-		$this->addPopulators(new NetherPopulator($world->getWorldHeight()));
+	public function __construct($seed, array $options = []){
+		parent::__construct($seed, Environment::NETHER, null, $options);
+		$this->addPopulators(new NetherPopulator($this->getWorldHeight())); // This isn't faithful to original code. Was $world->getWorldHeight()
```
Comment (Muqsit) [inline src/muqsit/vanillagenerator/generator/nether/NetherGenerator.php:46]: Why was this changed?

File: src/muqsit/vanillagenerator/generator/biomegrid/RiverMapLayer.php (hunk 2 above)
Comment (artulloss) [reply src/muqsit/vanillagenerator/generator/biomegrid/RiverMapLayer.php:13]: It looks like the difference with Pocketmines is that they have MOUNTAINS as 3, which is EXTREME_HILLS in your id constants as well as SWAMP being SWAMPLANDS. You want me to add pocketmines constants to yours and stick with yours?

This is my change in BiomeIds.php

```suggestion
public const EXTREME_HILLS, MOUNTAINS = 3;
	public const FOREST = 4;
	public const TAIGA = 5;
	public const SWAMPLAND, SWAMP = 6;
```

Comment (Muqsit) [reply src/muqsit/vanillagenerator/generator/biomegrid/RiverMapLayer.php:13]: Hmm, sounds like a bug. I imported the biome ids that MCPC has, which probably differs from the ones MCPE uses.

Comment (Muqsit) [reply src/muqsit/vanillagenerator/generator/biomegrid/RiverMapLayer.php:13]: I'd suggest something like
```php
namespace muqsit\vanillagen...;

use pocketmine\data\bedrock\BiomeIds as VanillaBiomeIds;

final class BiomeIds{

	public const OCEAN = VanillaBiomeIds::OCEAN;
}
```

File: src/muqsit/vanillagenerator/generator/nether/NetherGenerator.php (hunk 3 above)
Comment (artulloss) [reply src/muqsit/vanillagenerator/generator/nether/NetherGenerator.php:46]: This was changed because otherwise it doesn't extend VanillaGenerator properly.
[https://github.com/adams-pm-plugins/VanillaGenerator/blob/611ab10950975da9ebec1b8af6c1223dcccea725/src/muqsit/vanillagenerator/generator/VanillaGenerator.php#L43](url)

File: src/muqsit/vanillagenerator/generator/VanillaGenerator.php (hunk 1 above)
Comment (artulloss) [reply src/muqsit/vanillagenerator/generator/VanillaGenerator.php:21]: Remove the method entirely, or just the comment- the method isn't called anymore.

File: src/muqsit/vanillagenerator/generator/nether/NetherGenerator.php (hunk 3 above)
Comment (Muqsit) [reply src/muqsit/vanillagenerator/generator/nether/NetherGenerator.php:46]: Hmm alright, while at it you may as well remove https://github.com/adams-pm-plugins/VanillaGenerator/blob/master/src/muqsit/vanillagenerator/generator/VanillaGenerator.php#L19-#L32

File: src/muqsit/vanillagenerator/generator/VanillaGenerator.php (hunk 1 above)
Comment (Muqsit) [reply src/muqsit/vanillagenerator/generator/VanillaGenerator.php:21]: Yeah, you may as well remove all of this entirely: https://github.com/adams-pm-plugins/VanillaGenerator/blob/master/src/muqsit/vanillagenerator/generator/VanillaGenerator.php#L19-#L32
